File: src/main/java/com/jonathankee/schema/Discussion.java
package com.jonathankee.schema;

import java.util.List;

public class Discussion {
    private int number;
    private int questionNumber;
    private String questionExam;
    private String selectedAnswer;
    private String text;
    private int upvote;

    public Discussion(int number, int questionNumber, String questionExam, String selectedAnswer, String text, int upvote) {
        this.number = number;
        this.questionNumber = questionNumber;
        this.questionExam = questionExam;
        this.selectedAnswer = selectedAnswer;
        this.text = text;
        this.upvote = upvote;
    }

    public Discussion() {
        this.number = 0;
        this.questionNumber = 0;
        this.questionExam = "";
        this.selectedAnswer = "";
        this.text = "";
        this.upvote = 0;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public void setQuestionNumber(int questionNumber) {
        this.questionNumber = questionNumber;
    }

    public void setQuestionExam(String questionExam) {
        this.questionExam = questionExam;
    }

    public void setSelectedAnswer(String selectedAnswer) {
        this.selectedAnswer = selectedAnswer;
    }

    public void setText(String text) {
        this.text = text;
    }

    public void setUpvote(int upvote) {
        this.upvote = upvote;
    }

    public int getNumber() {
        return number;
    }

    public int getQuestionNumber() {
        return questionNumber;
    }

    public String getQuestionExam() {
        return questionExam;
    }

    public String getSelectedAnswer() {
        return selectedAnswer;
    }

    public String getText() {
        return text;
    }

    public int getUpvote() {
        return upvote;
    }

    @Override
    public String toString() {
        return "Dicussion{" +
                "number=" + number +
                ", questionNumber=" + questionNumber +
                ", questionExam='" + questionExam + '\'' +
                ", selectedAnswer='" + selectedAnswer + '\'' +
                ", text='" + text + '\'' +
                ", upvote=" + upvote +
                '}';
    }

    public static String insertMultiple(List<Discussion> dicussions) {
        String insertBoilerPlate = """
                    INSERT INTO discussions
                    (number, question_number, question_exam, selected_answer, text, upvote)
                    VALUES
                    """;

        List<String> data = dicussions.stream().map(dicussion -> {
            // Replace ' with '' so SQL syntax doesn't break
            String safeText1 = dicussion.getQuestionExam() != null ? dicussion.getQuestionExam().replace("'", "''") : "";
            String safeText2 = dicussion.getText() != null ? dicussion.getText().replace("'", "''") : "";
            String safeText3 = dicussion.getSelectedAnswer() != null ? dicussion.getSelectedAnswer().replace("'", "''") : "";

            return "(" + dicussion.getNumber() + ","
                    + dicussion.getQuestionNumber() + ","
                    + "'" + safeText1 + "'" + ","
                    + "'" + safeText2 + "'" + ","
                    + "'" + safeText3 + "'" + ","
                    + dicussion.getUpvote() + ")";
        }).toList();

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < data.size(); i++) {
            if (i != data.size() - 1) {
                sb.append(data.get(i) + ",\n");
            } else
                sb.append(data.get(i) + ";");
        }

        String fullInsert = insertBoilerPlate + sb.toString();
        return fullInsert;
    }
}

File: KotlinMain/src/main/kotlin/com.jonathankee/schema/Discussion.kt
package com.jonathankee.schema

class Discussion {
    var number: Int
    var questionNumber: Int
    private var questionExam: String
    private var selectedAnswer: String
    private var text: String
    var upvote: Int

    constructor(
        number: Int,
        questionNumber: Int,
        questionExam: String,
        selectedAnswer: String,
        text: String,
        upvote: Int
    ) {
        this.number = number
        this.questionNumber = questionNumber
        this.questionExam = questionExam
        this.selectedAnswer = selectedAnswer
        this.text = text
        this.upvote = upvote
    }

    constructor() {
        this.number = 0
        this.questionNumber = 0
        this.questionExam = ""
        this.selectedAnswer = ""
        this.text = ""
        this.upvote = 0
    }

    fun setQuestionExam(questionExam: String) {
        this.questionExam = questionExam
    }

    fun setSelectedAnswer(selectedAnswer: String) {
        this.selectedAnswer = selectedAnswer
    }

    fun setText(text: String) {
        this.text = text
    }

    fun getQuestionExam(): String? {
        return questionExam
    }

    fun getSelectedAnswer(): String? {
        return selectedAnswer
    }

    fun getText(): String? {
        return text
    }

    override fun toString(): String {
        return "Dicussion{" +
                "number=" + number +
                ", questionNumber=" + questionNumber +
                ", questionExam='" + questionExam + '\'' +
                ", selectedAnswer='" + selectedAnswer + '\'' +
                ", text='" + text + '\'' +
                ", upvote=" + upvote +
                '}'
    }

    companion object {
        fun insertMultiple(dicussions: List<Discussion>): String {
            val insertBoilerPlate = """
                    INSERT INTO scrape.discussions
                    (number, question_number, question_exam, selected_answer, text, upvote)
                    VALUES
                    
                    """.trimIndent()

            val data = dicussions.stream().map { dicussion: Discussion ->
                // Replace ' with '' so SQL syntax doesn't break
                val safeText1 = if (dicussion.getQuestionExam() != null) dicussion.getQuestionExam()!!
                    .replace("'", "''") else ""
                val safeText2 = if (dicussion.getText() != null) dicussion.getText()!!.replace("'", "''") else ""
                val safeText3 = if (dicussion.getSelectedAnswer() != null) dicussion.getSelectedAnswer()!!
                    .replace("'", "''") else ""
                ("(" + dicussion.number + ","
                        + dicussion.questionNumber + ","
                        + "'" + safeText1 + "'" + ","
                        + "'" + safeText2 + "'" + ","
                        + "'" + safeText3 + "'" + ","
                        + dicussion.upvote + ")")
            }.toList()

            val sb = StringBuilder()
            for (i in data.indices) {
                if (i != data.size - 1) {
                    sb.append(data[i] + ",\n")
                } else sb.append(data[i] + ";")
            }

            val fullInsert = insertBoilerPlate + sb.toString()
            return fullInsert
        }
    }
}